### ISSUE TITLE
[QUESTION]: V2: Why does MolGraphDataLoader default to shuffle = True?

### ISSUE THREAD

**Issue by KnathanM** [chemprop/chemprop#661]
Torch [Dataloader](https://pytorch.org/docs/stable/data.html#torch.utils.data.DataLoader) defaults to `shuffle = False`, but our `MolGraphDataLoader` [switches](https://github.com/chemprop/chemprop/blob/v2/dev/chemprop/data/dataloader.py#L37) the default to True. Was there a reason for this?

**Comment by davidegraff** [comment]
FWIW, the whole object can be removed and we can switch to just using a native `torch.data.DataLoader`

**Comment by KnathanM** [comment]
We could remove `MolGraphDataLoader`, but I've viewed it more as a helper wrapper around the native torch Dataloader. It helps automate the sampler so that users can get deterministic shuffling/sampling. We also added a warning message to it to help users navigate a limitation of batch normalization that you can't have a batch size of 1 (a dataset of size 101 would have a final batch of 1 if the batch size is 50).

If this class is just a helper wrapper, I think we should also set a default seed and always pass a sampler to the torch Dataloader. If the user doesn't want to use a sampler, they can just use the native torch Dataloader as you pointed out. 

I'm also satisfied to leave the default shuffle as True, based on my assumption that it is worse if the user should shuffle (training) and doesn't than if the user shouldn't shuffle (testing) and does. Our helper wrapper doesn't have the same signature as the native torch Dataloader, so a user would be wise to not assume they have the same defaults.

**Comment by KnathanM** [comment]
Correction: If we always pass a sampler then you have to shuffle. To give the user the option to not shuffle, this should keep the `sampler = None` logic. 

But we can still set a default seed. That way the seed for data is used for data splitting and loader sampling while the pytorch seed is used for model initialization. (Open discussion still about whether to keep these separate or combine them. They are separate for now.)

**Comment by kevingreenman** [comment]
note that the default `seed` of `MolGraphDataLoader` is still `None` and was not changed by #684

**Comment by KnathanM** [comment]
#755 Is also about making Chemprop deterministic. I found that we don't have to set the seed for `MolGraphDataLoader` because it pull randomness from the pytorch RNG backend. If we set the pytorch seed, the shuffling will be deterministic, which if `Trainer(deterministic=True)` is set, then training will be deterministic. 

If a user wants, they can set the seed for the shuffling manually, but we don't need to expose this through the CLI.
